### PR TITLE
More portable cades tweaks and buffs

### PR DESCRIPTION
**Pull request by ihatethisengine**
# About the pull request

Despite making a lot of tweaks and changes to portable cades I barely touched them in the game until recently. Now I have more experience and can tweak it again.

1) You can now stack damaged cades and stack stores health of each cade. You can repair stacked cades but it will take longer time.

2) Miniengi pamphlet allows faster repairs but only when cade is folded. 

3) You can quickly collapse portable cades with crowbar if you have at least miniengi skills.

4) You no longer need to have folded portable cade in hand in order to repair it, but if you do, you can move while repairing.

# Explain why it's good for the game

1) It's extremely annoying to repair each individual cade in order to stack them just because it got hit by a stray bullet once. Now you can just ignore damage and keep going.

2) Yeah it took 10 second for PFC to repair each single cade. Really long. Now it's 5 seconds, but only when folded and if you have miniengi skills. Makes miniengi pamphlet a bit more relevant.

3) It was intended, but turned out it was a bit inconvenient. It was faster to collapse by hand because you didn't need to deal with tools. Now it requires just a crowbar and slightly faster. Also requires miniengi skill to use crowbar.

4) First was just a bit annoying, second allows more mobility which is the point of portable barricades.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl: ihatethisengine
balance: you can stack scratched portable cades if their HP at least 75%. Doing so will reduce the health of all barricades in the stack to the level of the most damaged.
balance: you can repair stacked portable cades but it will take longer time depending on how many cades in stack.
balance: miniengi skill makes repairs of folded portable cades faster (10 seconds to 5 seconds, same as engineer).
balance: with engineering skill at least of miniengi you can collapse deployed portable barricade with a crowbar (wrench is no longer required, slightly faster (2 sec to 1.5 sec)).
balance: you no longer need to have folded portable cade in hand in order to repair it.
balance: if you have folded portable cade in hand, you can move while repairing it.
/:cl:

### CODE REVIEW

**Review by harryob — APPROVED** [review]
code is fine

**Comment by morrowwolf** [comment]
>You can now stack scratched portable cades if their HP > 75%. But this will reduce the health of all barricades in the stack to the level of the most damaged. You can repair stacked cades but it will take longer time.

This smells like a coder block rather than a real thought out feature. Make the stack store the damage values of the portacades or don't do this at all.

>Miniengi pamphlet allows faster repairs but only when cade is folded.

Why just when folded?

**Comment by ihatethisengine** [comment]
> Why just when folded?
Don't want to affect actual combat balance, because miniengi pamphlet doesn't allow faster repairs of regular cades. But if you think it's fine then sure, I'll let miniengi to repair unfolded portable cades faster.

**Comment by morrowwolf** [comment]
> > Why just when folded?
> 
> Don't want to affect actual combat balance, because miniengi pamphlet doesn't allow faster repairs of regular cades. But if you think it's fine then sure, I'll let miniengi to repair unfolded portable cades faster.

Honestly was just curious. This is good reasoning I say keep it.

**Comment by ihatethisengine** [comment]
> This smells like a coder block rather than a real thought out feature. Make the stack store the damage values of the portacades or don't do this at all.

Quite a task it was, but I think I got it working.

File: code/game/objects/structures/barricade/deployable.dm [hunk 1]
```diff
@@ -133,6 +115,8 @@
 	)
 	icon = 'icons/obj/items/marine-items.dmi'
 
+	var/list/stack_health = list(350)
```
Comment (harryob) [inline code/game/objects/structures/barricade/deployable.dm:118]: use initial to get the default health here, not hardcoded. you’ll probably need to do it in init

File: code/game/objects/structures/barricade/deployable.dm [hunk 2]
```diff
@@ -178,84 +162,98 @@
...
-/obj/item/stack/folding_barricade/attackby(obj/item/W, mob/user)
-	if(istype(W, /obj/item/stack/folding_barricade))
-		var/obj/item/stack/folding_barricade/F = W
-
-		if(health != maxhealth || F.health != F.maxhealth)
-			to_chat(user, "You cannot stack damaged [src.singular_name]\s.")
+/obj/item/stack/folding_barricade/attackby(obj/item/item, mob/user)
+	if(istype(item, /obj/item/stack/folding_barricade))
+		var/obj/item/stack/folding_barricade/folding = item
+		if(!ismob(src.loc)) //gather from ground
```
Comment (harryob) [inline code/game/objects/structures/barricade/deployable.dm:176]: src is implicit just use loc

File: code/game/objects/structures/barricade/deployable.dm [hunk 3]
```diff
@@ -277,8 +275,9 @@
 
 /obj/item/stack/folding_barricade/get_examine_text(mob/user)
 	. = ..()
-	if(health < maxhealth)
+	if(round(min(stack_health)/maxhealth * 100) <= 75)
 		. += SPAN_WARNING("It appears to be damaged.")
 
 /obj/item/stack/folding_barricade/three
 	amount = 3
+	stack_health = list(350, 350, 350)
```
Comment (harryob) [inline code/game/objects/structures/barricade/deployable.dm:283]: same here, you’ll probably need to do it in initialize though

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]

**Pull request merged**